### ISSUE TITLE
Inconsistency in Type Inference with zod + react-hook-form handleSubmit Validation Flow

### ISSUE THREAD

**Issue by Wisesaturn** [react-hook-form/react-hook-form#12446]
### Version Number

7.43.5

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/hgxggr

### Steps to reproduce

1. Install and set up react-hook-form and zod in your project
2. Create a form schema using zod that includes a transform
3. Initialize react-hook-form with zodResolver using the schema
4. Create a submit handler function
5. Check the console output when submitting the form
6. Observe that the runtime data structure doesn't match the TypeScript type inference

### Expected behavior:
- Type inference should match the actual runtime data structure (transformed output)

### Actual behavior:
- Type inference only considers the input type, not accounting for schema transformations

---

Hello, I'm a frontend developer implementing form validation using react-hook-form and zod. I've encountered an interesting behavior when using these together with zodResolver that I'd like to share.

Consider the following code:
```typescript
import "./styles.css";
import { useState } from "react";
import { z } from "zod";
import { useForm } from "react-hook-form";
import { zodResolver } from "@hookform/resolvers/zod";

const Schema = z.object({
  key: z
    .object({
      values: z.array(z.string()),
      input: z.string(),
    })
    .transform(({ values }) => values),
});

export default function App() {
  const [data, setData] = useState("none");
  const { handleSubmit } = useForm<z.input<typeof Schema>>({
    mode: "onChange",
    defaultValues: {
      key: {
        values: ["test", "test2"],
        input: "welcome",
      },
    },
    resolver: zodResolver(Schema),
  });

  const submitFunc = (data: z.input<typeof Schema>) => {
    // 🚨 Error!
    // real data is z.output<typeof Schema>
    // but type should use z.input<typeof Schema>;
    setData(data);
  };

  return (
    <form
      style={{ display: "flex", gap: "1rem", flexDirection: "column" }}
      onSubmit={handleSubmit(submitFunc)}
    >
      <input />
      {JSON.stringify(data)}
      <button type="submit">submit</button>
    </form>
  );
}
```

In this setup, zodResolver first performs validation and then applies the transform defined in the zod Schema, resulting in transformed data being passed to submitFunc. However, I have a question about the type inference:
Even though zodResolver applies the transform after validation, and the actual data reaching submitFunc is the transformed output, the type system only infers the input type (z.input<typeof Schema>). This seems counterintuitive since the actual runtime value matches the output type.

I'm curious about why react-hook-form's type system specifically enforces the input type for the submit handler, even though the actual data has been transformed by the time it reaches the handler. Is this a design decision or potentially an area where the type inference could be improved?

### What browsers are you seeing the problem on?

Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
```
export function useForm<
  TFieldValues extends FieldValues = FieldValues,
  TContext = any,
  TTransformedValues extends FieldValues | undefined = undefined,
>```

you will need to use `TTransformedValues`

**Comment by jhsong-muhayu** [comment]
it doesn't have `TTransformedValues` in `7.43.5`
I resolved this issue by upgrading to version 7.53.2, which includes TTransformedValues.
